Lib/http/server.py

```diff
@@ -892,15 +893,10 @@ def guess_type(self, path):
 
         """
         base, ext = posixpath.splitext(path)
-        if ext in self.extensions_map:
-            return self.extensions_map[ext]
-        ext = ext.lower()
-        if ext in self.extensions_map:
-            return self.extensions_map[ext]
-        guess, _ = mimetypes.guess_type(path)
-        if guess:
-            return guess
-        return 'application/octet-stream'
+        guess = self.extensions_map.get(ext,
+                self.extensions_map.get(ext.lower(),
+                mimetypes.guess_type(path)[0]))
```

Comment:
It is inefficient. First it call `guess_type()`, then `extensions_map.get(ext.lower())`, then `extensions_map.get(ext)`, while in many cases it could stop after `extensions_map.get(ext)`.

Reply:
Wow, you're right. I always assumed `dict.get('something', default)` short-circuited `default` on finding `'something'`. I guess just leave this part the way it was, then.

Reply:
As for "simpler to configure [my] browser", no. The big problem is with URLs with no extension, which was the majority of those captured by mitmproxy. And I couldn't just add ".html" to the saved files, because then the URLs attempting to access those pages wouldn't have worked.